src/lib.rs

```diff
@@ -462,30 +462,37 @@ fn median_select(data: &Vec<f64>, k: usize) -> Option<f64> {
     }
 }
 
-
 /*============================================ MAIN OPERATIONS : push, append, remove, pop ============================================= */
 impl SimpleAccumulator {
-    /// Similar to `push` in vector `Vec`. When `fixed_capacity` flag is 'true' and ring buffer is full
-    /// rewrites the oldest element with the latest, following FIFO order. 
+    /// Function similar to `push` in vector `Vec`. When `fixed_capacity` is 'true' and ring buffer is full,
+    /// the function rewrites the oldest element with the latest, following FIFO order.
+    /// 3 different cases arise when:
+    /// 1. capacity is not fixed,
+    /// 2. capacity is fixed but the buffer is not full and
+    /// 3. the buffer has fixed capacity and is full.
+    /// In the first two cases, the native push function for vectors is used to add the new element, all stats are updated
+    /// online and the number of data points incremented.
+    /// In the third case, we replace the oldest element by the new element (FIFO order). All stats are updated online
+    /// and the number of elements remains equal to the buffer capacity.
+    ///
     pub fn push<T: ToPrimitive>(&mut self, value: T) {
         let y = T::to_f64(&value).unwrap();
-        //let n = self.len as f64;
+
         // Running stats, Number of elements seen is incremented irrespective of buffer properties
         // Calculation is online following Knuth's algorithm
-        self.r_total += 1;
-        let r_delta = y - self.r_mean;
-        let r_delta_n = r_delta / self.r_total as f64;
-        self.r_mean += r_delta_n;
-        let r_term1 = r_delta * r_delta_n * (self.r_total as f64 - 1.0);
-        let r_stats1 = self.r_variance * (self.r_total as f64 - 2.0) + r_term1;
-        self.r_variance = r_stats1 / (self.r_total as f64 - 1.0);
+        self.total += 1;
+        let delta = y - self.mean;
+        let delta_n = delta / self.total as f64;
```

Comment:
This is wrong I think. `delta / self.total as f64` is `(delta / self.total) as f64` which is not `(delta as f64) / (self.total as f64)`.  Both of these are very different.